workspaces/src/operations.rs

```diff
@@ -393,3 +433,90 @@ impl<'a, 'b> CreateAccountTransaction<'a, 'b> {
         })
     }
 }
+
+/// `TransactionStatus` object relating to an [`asynchronous transaction`] on the network.
+/// Used to query into the status of the Transaction for whether it has completed or not.
+///
+/// [`asynchronous transaction`]: https://docs.near.org/api/rpc/transactions#send-transaction-async
+#[must_use]
+pub struct TransactionStatus<'a> {
+    client: &'a Client,
+    sender_id: AccountId,
+    hash: CryptoHash,
+}
+
+impl<'a> TransactionStatus<'a> {
+    pub(crate) fn new(
+        client: &'a Client,
+        id: AccountId,
+        hash: near_primitives::hash::CryptoHash,
+    ) -> Self {
+        Self {
+            client,
+            sender_id: id,
+            hash: CryptoHash(hash.0),
+        }
+    }
+
+    /// Checks the status of the transaction. If an [`TransactionPoll::Pending`] is returned, then the
+    /// transaction has not completed yet.
+    pub async fn status(&self) -> TransactionPoll<ExecutionFinalResult> {
```

Comment:
I wonder if using `Result<std::task::Poll<T>>` instead of a custom type might be preferred for ergonomics?

This seems like it might not be that ergonomic to use because they would have to match on the error if they wanted to just fail if it's hit.

What's the reason for introducing the new type `TransactionPoll`?

Reply:
`std::task::Poll` didn't have an error vairant, so went with a custom type, but then again, `Poll` has a couple impls that are nice to have and we don't repeat ourselves. Switched `Result<std::task::Poll<T>>` instead. Thanks for the suggestion